eth_abi/encoding.py

```diff
@@ -53,6 +53,15 @@ def encode(self, value):  # pragma: no cover
         """
         pass
 
+    def validate_value(self, value):
+        """
+        Determines whether or not the given value can be encoded by this
+        encoder.  Encoder classes can implement validation of values if desired.
+        By default, this method does nothing and an encoder will attempt to
+        encode any value it is given.
+        """
+        pass
```

Comment:
I mostly did it this way to expedience since the casper people need something quickly and there were some encoders that didn't validate their input values.  So it didn't seem like we'd break the behavior of the library significantly by allowing a default no-op.  But actually I think there are no encoders at this point which lack a `validate_value` method so perhaps we can just make it an abstract method like you say.